[2025-05-13T21:25:53.038+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: daily_stock_news_and_sentiment_dag.fetch_stock_prices scheduled__2025-05-12T00:30:00+00:00 [queued]>
[2025-05-13T21:25:53.050+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: daily_stock_news_and_sentiment_dag.fetch_stock_prices scheduled__2025-05-12T00:30:00+00:00 [queued]>
[2025-05-13T21:25:53.051+0000] {taskinstance.py:1283} INFO - 
--------------------------------------------------------------------------------
[2025-05-13T21:25:53.051+0000] {taskinstance.py:1284} INFO - Starting attempt 1 of 2
[2025-05-13T21:25:53.051+0000] {taskinstance.py:1285} INFO - 
--------------------------------------------------------------------------------
[2025-05-13T21:25:53.067+0000] {taskinstance.py:1304} INFO - Executing <Task(PythonOperator): fetch_stock_prices> on 2025-05-12 00:30:00+00:00
[2025-05-13T21:25:53.076+0000] {standard_task_runner.py:55} INFO - Started process 3228 to run task
[2025-05-13T21:25:53.079+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'daily_stock_news_and_sentiment_dag', 'fetch_stock_prices', 'scheduled__2025-05-12T00:30:00+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/daily_news_sentiment_dag.py', '--cfg-path', '/tmp/tmp_melazlt']
[2025-05-13T21:25:53.086+0000] {standard_task_runner.py:83} INFO - Job 4: Subtask fetch_stock_prices
[2025-05-13T21:25:53.217+0000] {task_command.py:389} INFO - Running <TaskInstance: daily_stock_news_and_sentiment_dag.fetch_stock_prices scheduled__2025-05-12T00:30:00+00:00 [running]> on host 3bd63daf2242
[2025-05-13T21:25:53.313+0000] {taskinstance.py:1511} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=daily_stock_news_and_sentiment_dag
AIRFLOW_CTX_TASK_ID=fetch_stock_prices
AIRFLOW_CTX_EXECUTION_DATE=2025-05-12T00:30:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2025-05-12T00:30:00+00:00
[2025-05-13T21:25:53.319+0000] {taskinstance.py:1772} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/daily_news_sentiment_dag.py", line 40, in fetch_stock_prices
    ts = TimeSeries(key=ALPHA_API_KEY, output_format='pandas')
  File "/home/airflow/.local/lib/python3.9/site-packages/alpha_vantage/alphavantage.py", line 50, in __init__
    raise ValueError('The AlphaVantage API key must be provided '
ValueError: The AlphaVantage API key must be provided either through the key parameter or through the environment variable ALPHAVANTAGE_API_KEY. Get a free key from the alphavantage website: https://www.alphavantage.co/support/#api-key
[2025-05-13T21:25:53.332+0000] {taskinstance.py:1322} INFO - Marking task as UP_FOR_RETRY. dag_id=daily_stock_news_and_sentiment_dag, task_id=fetch_stock_prices, execution_date=20250512T003000, start_date=20250513T212553, end_date=20250513T212553
[2025-05-13T21:25:53.345+0000] {standard_task_runner.py:100} ERROR - Failed to execute job 4 for task fetch_stock_prices (The AlphaVantage API key must be provided either through the key parameter or through the environment variable ALPHAVANTAGE_API_KEY. Get a free key from the alphavantage website: https://www.alphavantage.co/support/#api-key; 3228)
[2025-05-13T21:25:53.388+0000] {local_task_job.py:159} INFO - Task exited with return code 1
[2025-05-13T21:25:53.416+0000] {taskinstance.py:2582} INFO - 0 downstream tasks scheduled from follow-on schedule check
